### PR TITLE
Use Tilt CI to run e2e tests on Github workflows

### PR DESCRIPTION
**Pull request by brojd**
# What this PR does

- Reuse Tiltfile from local environment and use `tilt ci` to run e2e tests on Github
- Use Playwright Docker image to get rid of installing Playwright browsers and system dependencies
- Use ubuntu-latest-16-cores runner for e2e tests job on CI

## Which issue(s) this PR fixes

Closes https://github.com/grafana/oncall/issues/4018

## Checklist

- [x] Unit, integration, and e2e (if applicable) tests updated
- [x] Documentation added (or `pr:no public docs` PR label added if not required)
- [x] Added the relevant release notes label (see labels prefixed w/ `release:`). These labels dictate how your PR will
    show up in the autogenerated release notes.

### CODE REVIEW

File: .github/workflows/e2e-tests.yml [hunk 1]
```diff
@@ -33,7 +33,7 @@ jobs:
     # default "ubuntu-latest" runners only provide 2 CPU cores + 7GB of RAM. this seems to lead to HTTP 504s from
     # the oncall backend, and hence, flaky tests. Let's use CI runners w/ more resources to avoid this (plus
     # this will allow us to run more backend containers and parralelize the tests)
-    runs-on: ubuntu-latest-8-cores
+    runs-on: ubuntu-latest-16-cores
```
Comment (joeyorlando) [inline .github/workflows/e2e-tests.yml:36]: this exists?

File: .github/workflows/e2e-tests.yml [hunk 2]
```diff
@@ -149,102 +125,47 @@ jobs:
...
-      # spit out the engine, celery, and grafana logs, if the the e2e tests have failed (or were flaky)
-      # can be helpful for debugging these tests
-      # GitHub Action reference: https://github.com/jupyterhub/action-k8s-namespace-report
-      - name: oncall-engine logs
-        if: always()
-        uses: jupyterhub/action-k8s-namespace-report@v1
-        with:
-          important-workloads: deploy/oncall-ci-engine
+      - name: Install Playwright deps
+        uses: docker://mcr.microsoft.com/playwright:next-jammy
```
Comment (joeyorlando) [inline .github/workflows/e2e-tests.yml:145]: nice 👍

File: .github/workflows/e2e-tests.yml [hunk 3]
```diff
@@ -149,102 +125,47 @@ jobs:
...
-        env:
-          BASE_URL: http://localhost:30002/grafana
-          ONCALL_API_URL: http://172.17.0.1:30001
-          GRAFANA_ADMIN_USERNAME: oncall
-          GRAFANA_ADMIN_PASSWORD: oncall
-          GRAFANA_EDITOR_USERNAME: editor
-          GRAFANA_EDITOR_PASSWORD: editor
-          GRAFANA_VIEWER_USERNAME: viewer
-          GRAFANA_VIEWER_PASSWORD: viewer
-          MAILSLURP_API_KEY: ${{ secrets.MAILSLURP_API_KEY }}
```
Comment (joeyorlando) [inline .github/workflows/e2e-tests.yml:207]: where're these env vars getting passed into the tests?

File: .github/workflows/e2e-tests.yml [hunk 4]
```diff
@@ -149,102 +125,47 @@ jobs:
...
-      - name: grafana logs
-        if: always()
-        uses: jupyterhub/action-k8s-namespace-report@v1
-        with:
-          important-workloads: deploy/oncall-ci-grafana
+      - name: Tilt CI
+        shell: bash
+        env:
+          GRAFANA_IMAGE_TAG: ${{ inputs.grafana-image-tag }}
+        run: tilt ci
```
Comment (joeyorlando) [inline .github/workflows/e2e-tests.yml:151]: we'd likely need a way here to conditionally run the "expensive" tests

File: .github/workflows/e2e-tests.yml (hunk 1 above)
Comment (brojd) [reply .github/workflows/e2e-tests.yml:36]: Yep:)

File: .github/workflows/e2e-tests.yml (hunk 4 above)
Comment (brojd) [reply .github/workflows/e2e-tests.yml:151]: Will do it, thank you

**Review by teodosii — APPROVED** [review]

**Review by maskin25 — APPROVED** [review]

File: .github/workflows/e2e-tests.yml [hunk 5]
```diff
@@ -95,42 +98,6 @@ jobs:
         working-directory: grafana-plugin
         run: yarn build:dev
 
-      - name: Set up Docker Buildx # We need this step for docker caching
```
Comment (joeyorlando) [inline .github/workflows/e2e-tests.yml:98]: @brojd I suppose the Tiltfile will already take care of building the backend Docker image?

**Review by joeyorlando — APPROVED** [review]

File: .github/workflows/e2e-tests.yml (hunk 5 above)
Comment (brojd) [reply .github/workflows/e2e-tests.yml:98]: Yeah, we have this part in Tiltfile:
```
# Build the image including frontend folder for pytest
docker_build_sub(
    "localhost:63628/oncall/engine:dev",
    context="./engine",
    cache_from=["grafana/oncall:latest", "grafana/oncall:dev"],
    ignore=["./test-results/", "./grafana-plugin/dist/", "./grafana-plugin/e2e-tests/"],
    child_context=".",
    target="dev",
    extra_cmds=["ADD ./grafana-plugin/src/plugin.json /etc/grafana-plugin/src/plugin.json"],
    live_update=[
        sync("./engine/", "/etc/app"),
        run(
            "cd /etc/app && pip install pip-tools && pip-sync",
            trigger="./engine/requirements.txt",
        ),
    ],
)
```

**Pull request merged**